vumi_message_store/tests/test_message_store.py

```diff
@@ -656,6 +575,69 @@ def setUp(self):
         self.bi_cache = self.store.batch_info_cache
         self.msg_helper = self.add_helper(MessageHelper())
 
+    @inlineCallbacks
+    def _create_inbound_message_sequence(self, timestamps=False,
+                                         addresses=False, reverse=False):
+        """
+        Create a sequence of 5 inbound messages in a new batch and add them to
+        the backend. Returns the batch id and the list of message keys.
+        """
+        batch_id = yield self.backend.batch_start()
+        all_keys = []
+        start = (datetime.utcnow().replace(microsecond=0) -
+                 timedelta(seconds=10))
+        for i in xrange(5):
+            timestamp = start + timedelta(seconds=i)
+            addr = "addr%s" % (i,)
+            msg = self.msg_helper.make_inbound(
+                "Message %s" % (i,), timestamp=timestamp, from_addr=addr)
+            yield self.backend.add_inbound_message(msg, batch_ids=[batch_id])
+
+            key_tuple = [msg["message_id"], ]
+            if timestamps:
+                key_tuple.append(format_vumi_date(timestamp))
+            if addresses:
+                key_tuple.append(addr)
+            all_keys.append(
+                tuple(key_tuple) if len(key_tuple) > 1 else key_tuple[0])
```

Comment:
All these conditionals add complexity to this helper, which can make it harder to understand exactly what the tests using it are doing. Maybe it would be better to return the message objects (or the complete tuples) and have the tests extract the bits they want and reverse if necessary?